ocaml/runtime4/major_gc.c

```diff
@@ -670,7 +672,7 @@ Caml_noinline static intnat do_some_marking
         continue;
       }
       scan = Op_val(block);
-      obj_end = scan + Wosize_hd(hd);
+      obj_end = scan + Scannable_wosize_hd(hd);
```

Comment:
This needs to account for the work skipped or the GC will run at the wrong rate. `work` needs to decrease by 1 for each word of heap that is encountered, whether scanned or not. So I think something like `work -= wosize - scannable_sz` needs to happen here. (Possibly this is wrong in runtime5 too? Haven't checked)

Reply:
I believe this handles it in runtime 5: https://github.com/ocaml-flambda/flambda-backend/blob/5c5fb708dd41f5ee9fe2cebec3b1212cb850cf48/ocaml/runtime/major_gc.c#L924. It looks like I dropped it when porting the diff.

I've now added it -- this is ready for another review.

Reply:
Looks good!